torch/distributed/optim/utils.py

```diff
@@ -24,6 +24,18 @@
     optim.Adamax: _FunctionalAdamax,
 }
 
+
+# Interface to insert a new functional optimizer to functional_optim_map
+# Example::
+#     >>> import the new functional optimizer
+#     >>> from xyz import fn_optimizer
+#     >>> from torch.distributed.optim.utils import register_functional_optim
+#     >>>     register_functional_optim(key,fn_optimizer)
+
+def register_functional_optim(key,optim):
```

Comment:
nit: I think this is what @rohan-varma meant. I apologize if you were already meaning to fix.
```suggestion
def register_functional_optim(key, optim):
```

Reply:
@awgu The optim class need not be part of torch.optim, it can be a custom optim too. So a KV pair made sense in this scenario. Let me know if there are other options

Reply:
I see. We should make this distinction explicit in the docstring because right now `functional_optim_map` follows the type convention that I mentioned above.

To check my understanding, this means that for custom optimizers (non-`torch.optim.Optimizer`), you will pass it to the `ZeroRedundancyOptimizer` constructor here
https://github.com/pytorch/pytorch/blob/ba27ee9e8fc57f7509d2bf0c0be73510802806c0/torch/distributed/optim/zero_redundancy_optimizer.py#L370
and everything should work as expected.

Reply:
Yep - something like a simple 'object' inheritance. We can enforce it torch.optim too if thats the preferred way.